contrib/connector/nvmeof/nvmeof_helper.go

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) 2019 Intel Corporation, Ltd. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nvmeof
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/opensds/opensds/contrib/connector"
+)
+
+const (
+	iniNvmePrefix = "nqn.ini."
+)
+
+// ConnectorInfo define
+type ConnectorInfo struct {
+	Nqn       string `mapstructure:"targetNQN"`     //NVMe subsystem name to the volume to be connected
+	TgtPort   string `mapstructure:"targetPort"`    //NVMe target port that hosts the nqn sybsystem
+	TgtPortal string `mapstructure:"targetIP"`      //NVMe target ip that hosts the nqn sybsystem
+	TranType  string `mapstructure:"transporType "` // Nvme transport type
+	HostNqn   string `mapstructure:"hostNqn"`       // host nqn
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////
+//      Refer some codes from: https://github.intel.com/yingxinc/cinder-rsd-os-brick    //
+//////////////////////////////////////////////////////////////////////////////////////////
+
+// GetInitiator returns all the Nvmeof UUID
+func GetInitiator() ([]string, error) {
+	res, err := connector.ExecCmd("dmidecode")
+	nqns := []string{}
+	if err != nil {
+		log.Printf("Unable to execute dmidecode,Error encountered gathering Nvmeof UUID: %v", err)
+		return nqns, nil
+	}
+
+	lines := strings.Split(string(res), "\n")
+	for _, l := range lines {
+		if strings.Contains(l, "UUID:") {
+			tmp := iniNvmePrefix + strings.Split(l, ":")[1]
+			nqns = append(nqns, tmp)
+			log.Printf("Found the following nqns: %s", nqns)
+			return nqns, nil
+		}
+	}
+	log.Printf("Can not find any nqn initiator")
+	return nqns, errors.New("Can not find any nqn initiator")
```

Comment:
The begin letter of error message should be lower case.

Reply:
You mean that the begin letter of all log information, including error, infor, ...,  should be lower case, right?

Reply:
Yes